sklearn/cluster/_hdbscan/_tree.pyx

```diff
@@ -468,7 +518,7 @@ cdef get_probabilities(cnp.ndarray hierarchy, dict cluster_map, cnp.ndarray labe
 
         cluster = cluster_map[cluster_num]
         max_lambda = deaths[cluster]
-        if max_lambda == 0.0 or not np.isfinite(lambda_array[n]):
+        if max_lambda == 0.0 or isinf(lambda_array[n]):
```

Comment:
Should this be?

```suggestion
        if max_lambda == 0.0 or not isinf(lambda_array[n]):
```

Reply:
The original checks that it is `not finite` rather than `not infinite`, so I think the current `isinf` is correct -- unless I have it further mixed up somehow?

Reply:
Ah I'm thinking about:

```suggestion
        if max_lambda == 0.0 or not isfinite(lambda_array[n]):
```

Because of `nan` behavior:

```python
not isfinite(NaN) == 1
isinf(NaN) == 0
```

Is `lambda_array[n]` never `nan` by construction?